lib/widgets/nine_tile_box.dart

```diff
@@ -0,0 +1,138 @@
+import 'package:flutter/widgets.dart' as widgets;
+import 'package:meta/meta.dart';
+import '../sprite.dart';
+import 'dart:ui';
+
+class _Painter extends widgets.CustomPainter {
+  final Image image;
+  final double tileSize;
+  final double destTileSize;
+
+  _Painter(
+      {@required this.image,
+      @required this.tileSize,
+      @required this.destTileSize});
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    final topLeftCorner =
+        Sprite.fromImage(image, x: 0, y: 0, width: tileSize, height: tileSize);
+    final topRightCorner = Sprite.fromImage(image,
+        x: tileSize * 2, y: 0, width: tileSize, height: tileSize);
+
+    final bottomLeftCorner = Sprite.fromImage(image,
+        x: 0, y: 2 * tileSize, width: tileSize, height: tileSize);
+    final bottomRightCorner = Sprite.fromImage(image,
+        x: tileSize * 2, y: 2 * tileSize, width: tileSize, height: tileSize);
+
+    final topSide = Sprite.fromImage(image,
+        x: tileSize, y: 0, width: tileSize, height: tileSize);
+    final bottomSide = Sprite.fromImage(image,
+        x: tileSize, y: tileSize * 2, width: tileSize, height: tileSize);
+
+    final leftSide = Sprite.fromImage(image,
+        x: 0, y: tileSize, width: tileSize, height: tileSize);
+    final rightSide = Sprite.fromImage(image,
+        x: tileSize * 2, y: tileSize, width: tileSize, height: tileSize);
+
+    final middle = Sprite.fromImage(image,
+        x: tileSize, y: tileSize, width: tileSize, height: tileSize);
+
+    // Middle
+    for (var y = destTileSize;
+        y < size.height - destTileSize;
+        y = y + destTileSize) {
+      for (var x = destTileSize;
+          x < size.width - destTileSize;
+          x = x + destTileSize) {
+        middle.renderRect(
+            canvas, Rect.fromLTWH(x, y, destTileSize, destTileSize));
```

Comment:
you definitely don't need to do this... I thought of this at first, but please take a look at my implementation. you can expand the middle block and render it once. each 9 tile is rendered once, expand either: not at all, only in one direction and, the middle, on both directions. it's more efficient.

also this doesn't seem to cover if the width is not a multiple of destTileSize

Reply:
Not really, I did thought on expanding the tile initially, but if I expand, a nine tile box sprite like this will not work (I know this sprite is ugly, I made it very fast, just to make my point :) )

![ntb](https://user-images.githubusercontent.com/835641/73403669-b1171e80-42ce-11ea-8c5d-036a7bafd825.png)

If the in game implementation of the nin tile box is expanding, I suggest changing it to work with repeating tiles, otherwise we are excluding the support for sprites with details on the middles, which is something very cool that I think we should support.

> also this doesn't seem to cover if the width is not a multiple of destTileSize
Abou that, I agree with you, but I have made several tests on the dashbook story of this widgets, and it worked everytime, but I will give a second check.

Reply:
I see your use case, however, it makes it much more complex to handle the case I just mentioned. like if the container is 2.5 tiles of width, which part of the tiles will you show? I believe your "tiling" idea would only work well for perfectly tile multiples, but it's better to have something that works on every sizing, right? not sure, but I think we should have both game and widget impl. consistent

Reply:
Fair enough, to be honest, I would like to keep my implementation, to enable that kind of sprite I mentioned before, but yeah, I can't find a good solution for cases when the dimensions are not multiplier of the tile size, I am gonna change to expand the middle.